ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/factory/FragmentationStrategyExecutorCreatorImpl.java

```diff
@@ -1,36 +1,67 @@
 import io.micrometer.observation.ObservationRegistry;
 import org.springframework.stereotype.Service;
 
-import java.util.concurrent.Executors;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 
 @Service
 public class FragmentationStrategyExecutorCreatorImpl implements FragmentationStrategyExecutorCreator {
 
 	private final FragmentRepository fragmentRepository;
 	private final ObservationRegistry observationRegistry;
-	private final MemberToFragmentRepository memberToFragmentRepository;
+	private final EventSourceService eventSourceService;
+	private final FragmentSequenceRepository fragmentSequenceRepository;
+	private final FragmentationStrategyCreator fragmentationStrategyCreator;
 
 	public FragmentationStrategyExecutorCreatorImpl(FragmentRepository fragmentRepository,
 			ObservationRegistry observationRegistry,
-			MemberToFragmentRepository memberToFragmentRepository) {
+			EventSourceService eventSourceService,
+			FragmentSequenceRepository fragmentSequenceRepository,
+			FragmentationStrategyCreator fragmentationStrategyCreator) {
 		this.fragmentRepository = fragmentRepository;
 		this.observationRegistry = observationRegistry;
-		this.memberToFragmentRepository = memberToFragmentRepository;
+		this.eventSourceService = eventSourceService;
+
+		this.fragmentSequenceRepository = fragmentSequenceRepository;
+		this.fragmentationStrategyCreator = fragmentationStrategyCreator;
 	}
 
 	public FragmentationStrategyExecutor createExecutor(ViewName viewName,
-			FragmentationStrategy fragmentationStrategy) {
+			ViewSpecification viewSpecification) {
+		final FragmentationStrategy fragmentationStrategy = fragmentationStrategyCreator
+				.createFragmentationStrategyForView(viewSpecification);
 		final var rootFragmentRetriever = new RootFragmentRetriever(fragmentRepository, observationRegistry);
 		return new FragmentationStrategyExecutor(viewName, fragmentationStrategy, rootFragmentRetriever,
-				observationRegistry, memberToFragmentRepository, Executors.newSingleThreadExecutor());
+				observationRegistry, createExecutorService(), eventSourceService, fragmentSequenceRepository);
+	}
+
+	/**
+	 * This ExecutorService has the following properties:
+	 * - single threaded: For the time being we process members single threaded to
+	 * ensure their order
+	 * and avoid possible state issues of fragmentisers
+	 * - blocking queue: Ingestion can go fast, we do not want the queue to fill up
+	 * with tasks and consume too
+	 * much memory. Tasks should be aware of this and using a loop with checks might
+	 * be a good idea.
+	 * - discard policy: When the queue is full, new tasks are discarded.
+	 */
+	private ExecutorService createExecutorService() {
+		return new ThreadPoolExecutor(1, 1,
+				0L, TimeUnit.MILLISECONDS,
+				new ArrayBlockingQueue<>(2, true), new ThreadPoolExecutor.DiscardPolicy());
```

Comment:
This the fragmentation service is poll based, it seems like this can be reverted to a singleThreadExecutor.